src/nupic/research/frameworks/htm/temporal_memory/sequence_memory_apical_tiebreak.py

```diff
@@ -0,0 +1,196 @@
+
+class SequenceMemoryApicalTiebreak(TemporalMemoryApicalTiebreak):
+    """
+    sequence memory with apical tiebreak, built on temporal memory with apical tiebreak.
+    """
+
+    def __init__(
+        self,
+        num_minicolumns=2048,
+        apical_input_size=0,
+        num_cells_per_minicolumn=32,
+        activation_threshold=13,
+        reduced_basal_threshold=13,
+        initial_permanence=0.21,
+        connected_permanence=0.50,
+        matching_threshold=10,
+        sample_size=20,
+        permanence_increment=0.1,
+        permanence_decrement=0.1,
+        basal_segment_incorrect_decrement=0.0,
+        apical_segment_incorrect_decrement=0.0,
+        max_synapses_per_segment=-1,
+        seed=42
+    ):
+
+        params = {
+            "num_minicolumns" : num_minicolumns,
+            "basal_input_size" : num_minicolumns * num_cells_per_minicolumn,
+            "apical_input_size" : apical_input_size,
+            "num_cells_per_minicolumn" : num_cells_per_minicolumn,
+            "activation_threshold" : activation_threshold,
+            "reduced_basal_threshold" : reduced_basal_threshold,
+            "initial_permanence" : initial_permanence,
+            "connected_permanence" : connected_permanence,
+            "matching_threshold" : matching_threshold,
+            "sample_size" : sample_size,
+            "permanence_increment" : permanence_increment,
+            "permanence_decrement" : permanence_decrement,
+            "basal_segment_incorrect_decrement" : basal_segment_incorrect_decrement,
+            "apical_segment_incorrect_decrement" : apical_segment_incorrect_decrement,
+            "max_synapses_per_segment" : max_synapses_per_segment,
+            "seed" : seed
+        }
+
+        super().__init__(**params)
+
+        self.previous_apical_input = torch.empty(0, dtype=int_type).to(device)
+        self.previous_apical_growth_candidates = torch.empty(
+            0, dtype=int_type
+        ).to(device)
+        self.previous_predicted_cells = torch.empty(0, dtype=int_type).to(device)
+
+    def reset(self):
+        """
+        clear all cell and segment activity.
+        """
+
+        super().reset()
+
+        self.previous_apical_input = self.previous_apical_input.new_empty((0,))
+        self.previous_apical_growth_candidates = \
+            self.previous_apical_growth_candidates.new_empty((0,))
+        self.previous_predicted_cells = self.previous_predicted_cells.new_empty((0,))
+
+    def compute(
+        self,
+        active_minicolumns,
+        apical_input=torch.Tensor().to(device),
+        apical_growth_candidates=None,
+        learn=True
+    ):
+        """
+        perform one timestep: activate the specified minicolumn using the predictions
+        from the previous timestep and then learn. form a new set of predictions
+        using the newly active cells and the apical input.
+
+        `active_minicolumns` (torch.Tensor) contains the active minicolumns.
```

Comment:
This looks good, if it becomes stable code we might want to standardize the docstring style to a single format. I am not a big fan of the style we current use in nupic.research, I personally prefer numpy style, but I agree that standardizing is a good idea.

Reply:
The current docstring style is the one recommended by https://peps.python.org/pep-0287/
I am open to change any standard docstring format that is well documented and supported by sphinx and readthedocs. See https://nupicresearch.readthedocs.io/en/latest/

Reply:
Might have been misunderstood here. I am not suggesting we change docstring format. In this comment, I was suggesting we change the docstrings in this PR to match the current standard used in the rest of the repo.